src/test/scala/uk/gov/nationalarchives/downloadfiles/ExternalServicesTest.scala

```diff
@@ -19,6 +19,24 @@ class ExternalServicesTest extends AnyFlatSpec with BeforeAndAfterEach with Befo
 
   val wiremockGraphqlServer = new WireMockServer(9001)
   val wiremockAuthServer = new WireMockServer(9002)
+  val wiremockKmsEndpoint = new WireMockServer(9004)
+
+  def stubKmsResponse(cipherText: String): StubMapping =
+    wiremockKmsEndpoint.stubFor(post(urlEqualTo("/"))
+      .withRequestBody(equalToJson(s"""{"CiphertextBlob":"$cipherText","EncryptionContext":{"LambdaFunctionName":"test-lambda-function"}}"""))
+      .willReturn(okJson(s"""{"Plaintext": "$cipherText"}"""))
```

Comment:
Just trying to wrap my head around how this works. It looks like it takes encrypted text (e.g. `aHR0cDovL2xvY2FsaG9zdDo5MDAyL2F1dGg=`) and returns the same encrypted text. Is that right? Shouldn't it be returning decrypted text?

I might not have the right mental mode of this yet , but is there some way to set up a single general Wiremock response that extracts the plain text from the JSON and just returns it? So you can put unencrypted values in the test resources?

Reply:
In this PR https://github.com/nationalarchives/tdr-aws-utils/pull/24 you can see what's happening. You have to base64 decode the encrypted variable before passing it to KMS.
Then with the response, the call to `.asUtf8String()` base64 decodes the response again. 

So the request sent to  wiremock will be base64 encoded, and the response needs to be encoded. The application roperties are just base64 encoded versions of the original properties. If you try to pass in the non encoded properties, or try to return non encoded properties in the response, you get errors about invalid base64 encoding. 

Because everything into and out of KMS needs to be base64 encoded, I can't see a way of having them in the test properties without it. In the real lambda, the properties will be base64 encoded so the test ones need to be the same way I think.
I'm not particularly happy about it this way but I can't see a way around it.

Reply:
Oh, so they're not encrypted in the test application.conf? They're just base64-encoded?

Still, is it possible to have a generic Wiremock stubbing method so that when you add a new parameter, you don't have to add it in two places - application.conf _and_  `stubKmsResponse`?

Reply:
I did try but the parameters need to decode to the correct value for the tests to pass, so `http://localhost:8001/1/testqueueinput` is a test queue that we send and receive messages from. If we have a generic wiremock method then we need to make sure it's returning the correct value for that parameter. 
The only thing I can think of is recursively going through every property in the application.conf file, checking which ones are base64 encoded (some, like the sqs endpoints aren't because these are hard coded in the main application) and setting up wiremock stubs for them but this seems like overkill and probably more confusing.
I don't think we're going to be adding new environment variables often enough for this to be a  huge problem.

Reply:
Maybe not, but it will be in every Lambda repo. 😢 

Not sure if I've misunderstood how the KMS endpoint works, or if I haven't explained my suggestion very well. Might be easier to talk about on a call? Let me know when you have five minutes.